File: lib/commonAPI/bluetooth/ext/platform/iphone/impl/BluetoothConnectionSingleton.m

#import "BluetoothConnectionSingleton.h"
#import "Bluetooth.h"

@implementation BluetoothConnectionSingleton



-(void) getIsAvailable:(id<IMethodResult>)methodResult {
    [methodResult setResult:[NSNumber numberWithBool:YES]];
}

-(void) getDeviceName:(id<IMethodResult>)methodResult {
    [methodResult setResult:[RhoBluetoothManager sharedInstance].deviceName];
}

-(void) setDeviceName:(NSString*)deviceName methodResult:(id<IMethodResult>)methodResult {
    [RhoBluetoothManager sharedInstance].deviceName = [deviceName copy];
}

-(void) getLastError:(id<IMethodResult>)methodResult {
    [methodResult setResult:ERROR_NO_ERRORS];
}

-(void) enableBluetooth:(id<IMethodResult>)methodResult {
    //nothing
}

-(void) disableBluetooth:(id<IMethodResult>)methodResult {
    [[RhoBluetoothManager sharedInstance] doBluetoothOff];
}

-(void) getConnectionByID:(NSString*)connectionID methodResult:(id<IMethodResult>)methodResult {
    [methodResult setResult:connectionID];
}

-(void) createConnection:(NSString*)role methodResult:(id<IMethodResult>)methodResult {
    [[RhoBluetoothManager sharedInstance] startConnect:methodResult];
}

-(void) createServerConnection:(id<IMethodResult>)methodResult {
    [[RhoBluetoothManager sharedInstance] startServerConnect:@"" callback:methodResult];
}

-(void) createClientConnection:(NSString*)serverName methodResult:(id<IMethodResult>)methodResult {
    [[RhoBluetoothManager sharedInstance] startClientConnect:[serverName copy] callback:methodResult];
}

-(void) stopCurrentConnectionProcess:(id<IMethodResult>)methodResult {
    [[RhoBluetoothManager sharedInstance] stop_current_connection];
}




@end

File: lib/commonAPI/bluetooth/ext/platform/android/src/com/rho/bluetoothconnection/BluetoothConnectionSingleton.java
package com.rho.bluetoothconnection;

import java.util.LinkedList;
import java.util.List;

import com.rhomobile.rhodes.api.IMethodResult;
import com.rhomobile.rhodes.bluetooth.RhoBluetoothManager;

class BluetoothConnectionSingleton extends BluetoothConnectionSingletonBase implements IBluetoothConnectionSingleton {
    public BluetoothConnectionSingleton(BluetoothConnectionFactory factory) {
        super();
    }

    public void getIsAvailable(IMethodResult result) {
        result.set(RhoBluetoothManager.is_bluetooth_available() != 0);
    }

    public void getDeviceName(IMethodResult result) {
        result.set(RhoBluetoothManager.get_device_name());
    }

    public void setDeviceName(String deviceName, IMethodResult result) {
        RhoBluetoothManager.set_device_name(deviceName);
    }

    public void getLastError(IMethodResult result) {
        result.set("OK");
    }

    public void enableBluetooth(IMethodResult result) {
        //NOTHING
    }

    public void disableBluetooth(IMethodResult result) {
        RhoBluetoothManager.off_bluetooth();
    }

    public void getConnectionByID(String connectionID, IMethodResult result) {
        result.set(connectionID);
    }

    public void createConnection(String role, IMethodResult result) {
        RhoBluetoothManager.create_session(role, result);
    }

    public void createServerConnection(IMethodResult result) {
        RhoBluetoothManager.create_custom_server_session("",result);
    }

    public void createClientConnection(String serverName, IMethodResult result) {
        RhoBluetoothManager.create_custom_client_session(serverName, result);
    }

    public void stopCurrentConnectionProcess(IMethodResult result) {
        RhoBluetoothManager.stop_current_connection_process();
    }


}
